server/handler.go

```diff
@@ -103,6 +106,41 @@ func (h *Handler) ComQuery(
 
 	var r *sqltypes.Result
 	var proccesedAtLeastOneBatch bool
+
+	rowchan := make(chan sql.Row)
+	errchan := make(chan error)
+	quit := make(chan struct{})
+
+	// This goroutine will be select{}ed giving a change to Vitess to call the
+	// handler.CloseConnection callback and enforcing the timeout if configured
+	go func() {
+		for {
+			select {
+			case <-quit:
+				return
+			default:
+			}
+			row, err := rows.Next()
+			if err != nil {
+				errchan <- err
+				return
+			}
+			rowchan <- row
+		}
+	}()
+
+	// Default (big) waitTime is 3600, but it wont matter if there is no timeout
+	// because the loop will just iterate again. If there is a timeout, it will
+	// be enforced to ensure that Vitess has a chance to call Handler.CloseConnection()
+	totalTime := 0 * time.Second
```

Comment:
what's total time used for?

The only place it seems to be used is to assign waitTime to totalTime (which will make it be always equal to waitTime)

Reply:
Removed now, but it was an accumulator, on the loop of that commit there was a `totalTime += waitTime` it the timer expired and it would be checked to not exceed the connection timeout. Now if there is a timeout `waitTime` defaults to it or one hour if not (the time in this second case won't matter since it'll just loop again until the usually huge OS socket timeout expires).